src/popup/modules/UserInterface.js

```diff
@@ -378,6 +378,10 @@ export function handleQrError(error) {
 function triggerFileSave(file, filename, requestDownloadPermissions) {
     const downloadPermissionGranted = browser.permissions.contains(DOWNLOAD_PERMISSION);
 
+    // Blur QR code so that it is clear that user did not give permission yet
+    qrCode.style.filter = "blur(5px)"; // add the blur effect
```

Comment:
At least the comment on this line is unnecessary IMHO. Its okay you could leave it, but have a look at the clean code prinicples. One big takeaway is that the code itself should be self-explanatory and structured in way that you actually don't need to comment everything. If the code itself shows what is done there is no advantage in adding a comment that says the same again. Instead such comments can even be bad as they “rot” and may become outdated.